### PR TITLE
feat(ProcurementOperations): Add Bank File Generators (NACHA, PositivePay, SWIFT MT101)

### PR DESCRIPTION
**Pull request by azaharizaman**
## Summary

This PR implements production-ready bank file generation for the ProcurementOperations orchestrator using the Strategy Pattern. It addresses **Item #1** from the Phase A-C gap analysis: Bank File Generators.

### 🏗️ Architecture

Uses **Strategy Pattern** with versioning support for extensible bank file generation:

```
BankFileGeneratorInterface
    ├── AbstractBankFileGenerator (shared logic)
    ├── NachaFileGenerator (ACH payments)
    ├── PositivePayGenerator (check fraud prevention)
    └── SwiftMt101Generator (international wires)
```

#### Generators (`src/Strategies/BankFile/`)
| File | Description |
|------|-------------|
| `BankFileGeneratorInterface.php` | Common contract for all generators |
| `AbstractBankFileGenerator.php` | Shared implementation (validation, encoding) |
| `NachaFileGenerator.php` | NACHA 2025 Operating Rules compliant ACH files |
| `PositivePayGenerator.php` | Multi-format positive pay (6 bank formats) |
| `SwiftMt101Generator.php` | SWIFT MT101 international wire format |
| `BankFileGeneratorFactory.php` | Factory with convenience methods |

#### Value Objects (`src/ValueObjects/`)
| File | Description |
|------|-------------|
| `NachaConfiguration.php` | Company ID, routing/account numbers, SEC code |
| `PositivePayConfiguration.php` | Bank format, account number, company name |
| `SwiftMt101Configuration.php` | Sender BIC, ordering customer details, charge codes |

#### DTOs (`src/DTOs/BankFile/`)
| File | Description |
|------|-------------|
| `BankFileResultInterface.php` | Common result contract |
| `AbstractBankFileResult.php` | Shared result implementation |
| `NachaFileResult.php` | NACHA-specific metrics (entry count, hash, totals) |
| `PositivePayResult.php` | Check count and issue date tracking |
| `SwiftMt101Result.php` | Message reference and transaction count |

#### Enums (`src/Enums/`)
| File | Description |
|------|-------------|
| `BankFileFormat.php` | NACHA_ACH, POSITIVE_PAY, SWIFT_MT101 |
| `NachaSecCode.php` | PPD, CCD, CTX |
| `PositivePayFormat.php` | STANDARD_CSV, BAI2, BOA, WELLS_FARGO, CHASE, CITI |

#### Exception (`src/Exceptions/`)
| File | Description |
|------|-------------|
| `UnsupportedBankFileFormatException.php` | Thrown for unsupported formats |

#### Tests (`tests/Unit/Strategies/BankFile/`)
| File | Test Cases |
|------|------------|
| `NachaFileGeneratorTest.php` | 20+ tests |
| `PositivePayGeneratorTest.php` | 20+ tests |
| `SwiftMt101GeneratorTest.php` | 20+ tests |
| `BankFileGeneratorFactoryTest.php` | 15+ tests |

### 📝 Modified Files

**`src/DTOs/Financial/PaymentItemData.php`**
- Added check-specific fields: `checkDate`, `checkType`
- Added international payment fields: `beneficiaryBic`, `beneficiaryIban`, `beneficiaryName`, `beneficiaryAddress`, `beneficiaryCountry`, `intermediaryBic`, `chargeCode`
- New factory method: `forInternationalWire()`

### NachaFileGenerator
- **NACHA 2025 Operating Rules** compliant
- 94-character fixed-width format with 10-record blocking
- Supports SEC codes: PPD (payroll), CCD (corporate), CTX (corporate tax)
- Proper header/batch/entry/addenda record generation
- Hash and entry count validation

### PositivePayGenerator
- **6 bank format support:**
  - Standard CSV
  - BAI2
  - Bank of America
  - Wells Fargo
  - Chase
  - Citi
- Check types: ISSUED, VOID, STOP, STALE
- Bank-specific field requirements and formatting

### SwiftMt101Generator
- **SWIFT MT101 Request for Transfer** format
- International wire payment support
- BIC validation and ordering customer formatting
- Charge code handling (SHA, OUR, BEN)
- Intermediary bank support

## Usage Example

```php
use Nexus\ProcurementOperations\Strategies\BankFile\BankFileGeneratorFactory;
use Nexus\ProcurementOperations\DTOs\Financial\PaymentBatchData;

// Create factory
$factory = new BankFileGeneratorFactory();

// Generate NACHA file for vendor payments
$nachaGenerator = $factory->createNachaForVendorPayments(
    companyId: '1234567890',
    companyName: 'ACME Corp',
    originatingDfi: '12345678',
    accountNumber: '123456789012'
);

$result = $nachaGenerator->generate($paymentBatch);

// Access file content and metadata
$fileContent = $result->getContent();
$entryCount = $result->getTotalEntryCount();
$debitTotal = $result->getTotalDebitAmount();
```

## Testing

- **75+ unit test cases** covering:
  - Normal file generation operations
  - Edge cases (empty batches, maximum values)
  - Format compliance validation
  - Error handling
  - Factory behavior

## Checklist

- [x] Follows Nexus CODING_GUIDELINES.md
- [x] Uses `declare(strict_types=1)` in all files
- [x] All classes are `final readonly` where appropriate
- [x] All dependencies injected via constructor as interfaces
- [x] Comprehensive unit tests
- [x] No framework dependencies (pure PHP 8.3+)
- [x] Strategy Pattern for extensibility
- [x] Factory Pattern for convenience

## Related Issues

Addresses Phase A-C Gap Analysis Item #1: Bank File Generators (NachaFileGenerator, PositivePayGenerator)

### CODE REVIEW

File: orchestrators/ProcurementOperations/src/ValueObjects/DuplicateMatch.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
  *
  * Contains details about the matching invoice and the type of match detected.
  */
-final readonly class DuplicateMatch
+final class DuplicateMatch
```
[the remark this answers is not in the document]
Comment (azaharizaman) [reply orchestrators/ProcurementOperations/src/ValueObjects/DuplicateMatch.php]: False positive
